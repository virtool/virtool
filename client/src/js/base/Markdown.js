import Marked from "marked";
import React from "react";
import styled from "styled-components";
import { NoneFound } from "./NoneFound";

const StyledMarkdown = styled.div`
    overflow-y: scroll;
    max-height: 375px;
    margin-bottom: 0px;
    padding: 10px 15px;
`;

export const Markdown = ({ markdown }) => {
    if (markdown) {
        return <StyledMarkdown dangerouslySetInnerHTML={{ __html: Marked(markdown) }} />;
    }

    return (
        <StyledMarkdown>
            <NoneFound noun="notes" />
        </StyledMarkdown>
    );
};
