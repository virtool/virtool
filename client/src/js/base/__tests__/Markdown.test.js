import React from "react";
import { Markdown } from "../Markdown";

describe("<Markdown />", () => {
    it("should render with markdown", () => {
        const markdown = `
            # Heading
            body
        `;
        const wrapper = shallow(<Markdown markdown={markdown} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render NoneFound without markdown", () => {
        const wrapper = shallow(<Markdown markdown="" />);
        expect(wrapper).toMatchSnapshot();
    });
});
