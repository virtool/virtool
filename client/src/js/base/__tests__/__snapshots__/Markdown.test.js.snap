// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Markdown /> should render NoneFound without markdown 1`] = `
<Markdown__StyledMarkdown>
  <NoneFound
    noun="notes"
  />
</Markdown__StyledMarkdown>
`;

exports[`<Markdown /> should render with markdown 1`] = `
<Markdown__StyledMarkdown
  dangerouslySetInnerHTML={
    Object {
      "__html": "<pre><code>        # Heading
        body</code></pre>
",
    }
  }
/>
`;
